Content.Shared/Chemistry/Components/Solution.cs

```diff
@@ -674,58 +675,97 @@ public Solution SplitSolution(FixedPoint2 toTake)
             return newSolution;
         }
 
+        // Frontier: cryogenics per-reagent filter function (#1443)
         /// <summary>
-        /// Frontier
-        /// splits the solution taking up to the specified amount of each reagent from the solution.
-        /// If the solution has less of a reagent than the specified amount, it will take all of that reagent.
+        /// Splits a solution, taking the specified amount of each reagent from the solution.
+        /// If any reagent in the solution has less volume than specified, it will all be transferred into the new solution.
         /// </summary>
-        /// <param name="toTakePer">How much of each reagent to take</param>
-        /// <returns>a new solution containing the reagents taken from the original solution</returns>
-        public Solution SplitSolutionReagentsEvenly(FixedPoint2 toTakePer)
+        /// <param name="toTakePer">How much of each reagent to take.</param>
+        /// <returns>A new solution containing the reagents taken from the original solution.</returns>
+        public Solution SplitSolutionPerReagent(FixedPoint2 toTakePer)
         {
-            var splitSolution = new Solution();
-
             if (toTakePer <= FixedPoint2.Zero)
-                return splitSolution;
-            var reagentsCount = Contents.Count;
-            var reagentsToRemove = new List<ReagentQuantity>();
-            for (var i = 0; i < reagentsCount; i++)
-            {
-                var currentReagent = Contents[i];
+                return new Solution();
 
-                if (currentReagent.Quantity <= FixedPoint2.Zero)
```

Comment:
this safety check existed in the original code for a reason, we should not remove it